usaspending_api/search/v2/views/spending_by_category_views/spending_by_industry_codes.py

```diff
@@ -84,25 +92,57 @@ def query_django_for_subawards(self, base_queryset: QuerySet) -> List[dict]:
         django_filters = {f"{self.industry_code_type.value}__isnull": False}
         django_values = [self.industry_code_type.value]
 
-        queryset = self.common_db_query(base_queryset, django_filters, django_values).annotate(
-            code=F(self.industry_code_type.value)
-        )
+        if self.industry_code_type == IndustryCodeType.DEFC:
+            django_filters = {f"award__disaster_emergency_fund_codes__isnull": False}
+            django_values = ["award__disaster_emergency_fund_codes"]
+            queryset = self.common_db_query(base_queryset, django_filters, django_values).annotate(
+                code=F("award__disaster_emergency_fund_codes")
+            )
+        else:
+            queryset = self.common_db_query(base_queryset, django_filters, django_values).annotate(
+                code=F(self.industry_code_type.value)
+            )
+
         lower_limit = self.pagination.lower_limit
         upper_limit = self.pagination.upper_limit
         query_results = list(queryset[lower_limit:upper_limit])
 
-        for row in query_results:
-            if self.industry_code_type == IndustryCodeType.CFDA:
-                row["id"], row["name"] = fetch_cfda_id_title_by_number(row["code"])
-            elif self.industry_code_type == IndustryCodeType.PSC:
-                row["id"] = None
-                row["name"] = fetch_psc_description_by_code(row["code"])
-            elif self.industry_code_type == IndustryCodeType.NAICS:
-                row["id"] = None
-                row["name"] = fetch_naics_description_from_code(row["code"], row.get("name"))
-            row.pop(self.industry_code_type.value)
-
-        return query_results
+        # Custom filter for DEFC
```

Comment:
In the current state, this special logic for DEFC seems to be executed regardless of the `industry_code_type`. Perhaps it should be moved into the `if self.industry_code_type == IndustryCodeType.DEFC:` condition below?

Reply:
Separated to it's own method due to codeclimate complaints - but agreed